src/index.js

```diff
@@ -19,30 +19,48 @@ var setup = state.current();
 
 // get into dom asap to trigger asset loading
 function renderHolder() {
-    var url = util.resolveClientUrl(util.currentRequestId(), true);
-    var html = '<div class="glimpse"><a class="glimpse-icon" target="_blank" href="' + url + '"><div class="glimpse-icon-text">Glimpse</div></a><div class="glimpse-hud"></div></div>'
-    $(html).appendTo('body');
+    if (document.readyState === 'complete') {
```

Comment:
I don't particularly like fixing bugs without understanding root cause, but if you're confident this fixes it, let's go ahead and land it.